client/src/elm/Pages/MessagingCenter/Utils.elm

```diff
@@ -108,11 +108,48 @@ surveyQuestionsAnswered surveyType surveyForm =
     Dict.size surveyForm == List.length surveyQuestions
 
 
-resolveSurveyScoreDialogState : ResilienceSurveyType -> Int -> SurveyScoreDialogState
-resolveSurveyScoreDialogState surveyType score =
+resolveSurveyScoreDialogState : NurseId -> ResilienceSurveyType -> Int -> ModelIndexedDb -> SurveyScoreDialogState
+resolveSurveyScoreDialogState nurseId surveyType score db =
     case surveyType of
         ResilienceSurveyQuarterly ->
             QuarterlySurveyScore score
 
         ResilienceSurveyAdoption ->
-            AdoptionSurveyScore score
+            let
+                surveys =
+                    Dict.get nurseId db.resilienceSurveysByNurse
+                        |> Maybe.andThen RemoteData.toMaybe
+                        |> Maybe.map Dict.values
+                        |> Maybe.withDefault []
+
+                adoptionSurveys =
+                    List.filter (\survey -> survey.surveyType == ResilienceSurveyAdoption) surveys
+
+                sortedSurveys =
+                    List.sortWith (\survey1 survey2 -> Date.compare survey1.dateMeasured survey2.dateMeasured) adoptionSurveys
+
+                toScore answer =
```

Comment:
This function repeats itself. Lets make it a utility function to prevent code duplication.